server/api/api.go

```
@@ -42,15 +42,18 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request, sourcePlugin
 }
 
 // ReturnJSON writes the given pointer to object as json with a success response
-func ReturnJSON(w http.ResponseWriter, pointerToObject interface{}) {
+func ReturnJSON(w http.ResponseWriter, pointerToObject interface{}, httpStatus int) {
 	jsonBytes, err := json.Marshal(pointerToObject)
 	if err != nil {
 		HandleError(w, errors.Wrapf(err, "unable to marshal json"))
 		return
 	}
 
-	w.WriteHeader(http.StatusOK)
-	_, _ = w.Write(jsonBytes)
+	w.WriteHeader(httpStatus)
+	if _, err = w.Write(jsonBytes); err != nil {
+		HandleError(w, err)
```

If this where to actually fail this would probably end up just failing again trying to do the w.Write inside HandleError.
I think we usually just ignore this error because if the write fails there isn't a whole lot we can do. 
We could just log an error here though.

---

You're right! Let me fix it.

---

I took a look and I think it's ok for use to `HandleError` here, as that function is already logging the error.

---

Ya. It's just a bit strange to call it here if that's the only thing we are getting from it. 0/5